File: Challenges/TwoSun/TwoSun.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public static void main(String[] args) {
  int[] array = { 2, 7, 11, 15};

  System.out.println(Arrays.toString(twoSun(array, 9)));

}

public static int[] twoSun(int[] nums, int target) {
  Map<Integer, Integer> hasher = new HashMap<>();

  for (int idx = 0; idx < nums.length; idx++) {
    int i = nums[idx];
    if (hasher.containsKey(i)) {
      return new int[] { hasher.get(i), idx };
    }

    hasher.put(target - i, idx);

  }

  return null;

}

File: Challenges/TwoSun/twosun.ts
const array = [3, 2, 4]
console.log(twoSun(array, 6))

function twoSun(nums: number[], target: number): number[]  {
  const hasher: { [key: number]: number } = {};

  for (let idx = 0; idx < nums.length; idx++) {
    const i = nums[idx]
    if (hasher[i] !== undefined) {
      return [hasher[i], idx]
    }

    hasher[target - 1] = idx

  }

  return [];

}